graphql/pupil/mutations.ts

```diff
@@ -58,4 +60,25 @@ export class MutatePupilResolver {
 
         return true;
     }
+
+    @Mutation(returns => Boolean)
+    @Authorized(Role.ADMIN)
+    async pupilCreateMatchRequest(@Arg("pupilId") pupilId: number, @Ctx() context: GraphQLContext): Promise<boolean> {
+        const pupil = await getPupil(pupilId);
+        const isAdmin = context.user.roles.includes(Role.ADMIN);
+
+        await createMatchRequest(pupil, isAdmin);
+
+        return true;
```

Comment:
What happens if `createMatchRequest` doesn't go through?

Reply:
Then it throws an error back to the caller (e.g. the Retool) which can show an error message to the user. Actually I wanted to return nothing here (a "void method"), but unfortunately a Void type is [not supported](https://stackoverflow.com/questions/44737043/is-it-possible-to-not-return-any-data-when-using-a-graphql-mutation). Thus I chose to return `true` which is somewhat similar. I'll add that to the documentation somewhen 😅

Reply:
Alright, it's fine. I just found the `return true;` a little suspicious 😅

Reply:
Yeah, would also prefer if TypeGraphQL would support it ...